lib/Doctrine/ODM/PHPCR/Mapping/ClassMetadataFactory.php

```diff
@@ -133,6 +133,10 @@ protected function doLoadMetadata($class, $parent, $rootEntityFound, array $nonS
             $this->getDriver()->loadMetadataForClass($class->getName(), $class);
         }
 
+        if ($parent && $class->getInheritMixins()) {
+            $class->setMixins(array_merge($parent->getMixins(), $class->getMixins()));
+        }
```

Comment:
adding inherited mapping happens before the class' metadata is loaded, so this needs to happen here

Cheers
Uwe

> Am 14.07.2015 um 08:25 schrieb David Buchmann <notifications@github.com>:
> 
> In lib/Doctrine/ODM/PHPCR/Mapping/ClassMetadataFactory.php:
> 
> > @@ -133,6 +133,10 @@ protected function doLoadMetadata($class, $parent, $rootEntityFound, array $nonS
> >              $this->getDriver()->loadMetadataForClass($class->getName(), $class);
> >          }
> >  
> > +        if ($parent && $class->getInheritMixins()) {
> > +            $class->setMixins(array_merge($parent->getMixins(), $class->getMixins()));
> > +        }
> i think this should be done inside addInheritedDocumentOptions
> 
> —
> Reply to this email directly or view it on GitHub.
>